File: C++/14. Longest Common Prefix.cpp
#include <bits/stdc++.h>

using namespace std;

string longestCommonPrefix(vector<string> &strs)
{
    sort(strs.begin(), strs.end());

    string result = "";
    string first = strs[0], last = strs.back();

    for (int i = 0; i < min(first.size(), last.size()); i++)
    {
        if (first[i] != last[i])
            break;
        result += first[i];
    }

    return result;
}

File: Java/14. Longest Common Prefix.java
import java.util.Arrays;

class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder result = new StringBuilder();
        Arrays.sort(strs);

        String a = strs[0];
        String b = strs[strs.length - 1];
        int n = Math.min(a.length(), b.length());

        for (int i = 0; i < n; i++) {
            if (a.charAt(i) != b.charAt(i))
                break;
            result.append(a.charAt(i));
        }

        return result.toString();
    }
}